backend/BMT-backend/Application/Queries/GetAllEnterprisesEarningsQuery.cs

```diff
@@ -0,0 +1,53 @@
+﻿using BMT_backend.Application.Services;
+using BMT_backend.Presentation.DTOs;
+using BMT_backend.Domain.Entities;
+
+namespace BMT_backend.Application.Queries
+{
+    public class GetAllEnterprisesEarningsQuery
+    {
+        private readonly EnterpriseService _enterpriseService;
+        private readonly GetEnterpriseEarningsQuery _getEnterpriseEarningsQuery;
+
+        public GetAllEnterprisesEarningsQuery(EnterpriseService enterpriseService, GetEnterpriseEarningsQuery getEnterpriseEarningsQuery)
+        {
+            _enterpriseService = enterpriseService;
+            _getEnterpriseEarningsQuery = getEnterpriseEarningsQuery;
+        }
+
+        public async Task<List<EarningsDatasetDto>> GetAllEnterprisesEarningsAsync()
+        {
+            try
+            {
+                List<EarningsDatasetDto> earnings = [];
+                List<Enterprise> enterprises = await _enterpriseService.GetAllEnterprisesAsync();
+                foreach (Enterprise enterprise in enterprises)
+                {
+                    List<EarningsDatasetDto> enterpriseProductEarnings = await _getEnterpriseEarningsQuery.GetEnterpriseEarningsAsync(enterprise.Id);
+                    EarningsDatasetDto enterpriseEarnings = new()
+                    {
+                        Label = enterprise.Name,
+                        EarningsPerMonth = new List<double>(new double[12])
+                    };
+                    for (int i = 0; i < 12; i++)
+                    {
+                        double EnterpriseMonthEarnings = 0;
+                        foreach (var productsEarnings in enterpriseProductEarnings)
+                        {
+                            EnterpriseMonthEarnings += productsEarnings.EarningsPerMonth[i];
+                        }
+                        enterpriseEarnings.EarningsPerMonth[i] = EnterpriseMonthEarnings;
+                    }
+                    earnings.Add(enterpriseEarnings);
+                }
+                return earnings;
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                throw new Exception("Error al obtener las ganancias de la empresa");
+            }
```

Comment:
El uso de try y catch debe ser considerado con base en la frecuencia en que una operación puede fallar.

Además, las excepciones tienen un impacto en el rendimiento generalizado de la operación, por lo que a nivel de arquitectura, se debería reconsiderar el uso de las excepciones como manejo de flujos de negocio, entre ellos, errores de aplicación particulares, por ejemplo, el obtener la información de la empresa.

Para más información, les sugiero visitar las buenas prácticas que el framework tiene documentadas

https://learn.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions